qualtran/bloqs/mod_arithmetic/mod_addition.py

```diff
@@ -87,13 +87,15 @@ def signature(self) -> 'Signature':
     def on_classical_vals(
         self, x: 'ClassicalValT', y: 'ClassicalValT'
     ) -> Dict[str, 'ClassicalValT']:
-        if not (0 <= x < self.mod):
+        # The construction still works when at most one of inputs equals `mod`.
+        special_case = (x == self.mod) ^ (y == self.mod)
```

Comment:
can we use `or` instead of `^`

Reply:
this is `at most` not `at least` ... although I can simply say `not (x == y == self.mod)` and update the inequalities below from `x < mod` to `x <= mod`, what do you prefer?